rocAL/rocAL_pybind/rocal_pybind.cpp

```diff
@@ -308,6 +303,270 @@ namespace rocal{
+                    case RocalTensorDataType::FP16:
+                        return "float16";
+                }
+            },
+                R"code(
+                Returns dtype of tensor.
+                )code"
+            )
+            .def("torch_dtype", [](rocalTensor &output_tensor) {
+                switch(output_tensor.info().data_type()) {
+                    case RocalTensorDataType::FP32:
+                        return "torch.float32";
+                    case RocalTensorDataType::UINT8:
+                        return "torch.float16";
+                    case RocalTensorDataType::FP16:
+                        return "torch.uint8";
+                }
+            },
+                R"code(
+                Returns dtype of torch tensor.
+                )code"
+            )
+            .def("numpy_dtype", [](rocalTensor &output_tensor) {
+                switch(output_tensor.info().data_type()) {
+                    case RocalTensorDataType::FP32:
+                        return "np.float32";
+                    case RocalTensorDataType::UINT8:
+                        return "np.float16";
+                    case RocalTensorDataType::FP16:
+                        return "np.uint8";
+                }
+            },
+                R"code(
+                Returns dtype of torch tensor.
+                )code"
+            )
+            .def("cupy_dtype", [](rocalTensor &output_tensor) {
+                switch(output_tensor.info().data_type()) {
+                    case RocalTensorDataType::FP32:
+                        return "cp.float32";
+                    case RocalTensorDataType::UINT8:
+                        return "cp.float16";
+                    case RocalTensorDataType::FP16:
+                        return "cp.uint8";
+                }
+            },
+                R"code(
+                Returns dtype of torch tensor.
+                )code"
+            )
+            .def("dims", [](rocalTensor &output_tensor) {
+                return output_tensor.info().dims();
+            },
+                R"code(
+                Returns dims of tensor.
+                )code"
+            )
+            m.def("dimensions", []=(rocalTensor &output_tensor) {
+            py::list list;
+            for (uint i = 0; i < output_tensor.info().dims(); i++)
+                list.append(output_tensor.info().dims().at(i));
+            return list; 
+            });
+            .def(
+            "copy_data", [](rocalTensor &output_tensor, py::object p) {
+                auto ptr = ctypes_void_ptr(p);
+                output_tensor.copy_data(ptr);
+            },
+                R"code(
+                Copies the ring buffer data to python buffer pointers.
+                )code"
+            )
+            .def(
+                "at",
+                [](rocalTensor &output_tensor, uint idx) {
+                    uint h = output_tensor.info().max_shape().at(1);
```

Comment:
please use max_shape directly instead of getting each element. Also I think it is better to keep max_shape and dims as arrays instead of std::vector for ease of access of these conversions

Reply:
Ideally it would be better to get only one call shape and dims and reuse it as needed

Reply:
Rewrote the code